analytics.js

```diff
@@ -1023,6 +1023,11 @@ class Analytics {
       }
     }
 
+    if (options && options.useBeacon === true) {
+      this.eventRepository.useBeacon = options.useBeacon;
+      this.eventRepository.initializeTransportMechanism();
+    }
```

Comment:
This should be done before queuing is started (Ln 985)

Reply:
This is used to clear the previous queue if any and destroy the queue. The name is not appropriate.

Reply:
Why are we starting the queue in the first place when the decision to use the beacon or not is not yet decided?

Reply:
I think there is some misunderstanding. Say, first time you load the page with useBeacon: true. That time a queue will be created with the timestamp. Then if you reload the page a new queue will be created. So to send data from the previous queue and destroy it this is necessary. 
I have updated the name to "initializeBeaconQueue".

Reply:
done